src/application.coffee

```diff
@@ -322,6 +331,28 @@ wrapAsError = (err) ->
 	return err if _.isError(err)
 	return new Error(err.message ? err)
 
+selectAndKill = (appId) ->
```

Comment:
This could be:
```coffee
	knex('app').select().where({ appId })
	.then ([ app ]) ->
		if !app?
			throw new Error('App not found')
		return app
```
and be used as `select(appId).then(kill)` or for https://github.com/resin-io/resin-supervisor/pull/8/files#diff-3e54a04880f9ac9cf4ef339b4c931ef6R392 as `select(localApp.appId).then(fetch)  if needsDownload`